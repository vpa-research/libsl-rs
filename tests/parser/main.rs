use std::fs;

use insta::{assert_ron_snapshot, Settings};
use libsl::{LibSl, WithLibSl};
use pathdiff::diff_paths;

#[test]
fn test_snapshots() {
    let mut settings = Settings::clone_current();
    settings.set_omit_expression(true);
    let _guard = settings.bind_to_scope();

    insta::glob!("test-data/**/*.libsl", |path| {
        let contents = fs::read_to_string(path)
            .map_err(|e| format!("could not read `{}` to string: {e}", path.display()))
            .unwrap();

        let mut settings = Settings::clone_current();
        settings.set_description(format!("// Input file:\n{contents}"));
        let _guard = settings.bind_to_scope();

        let mut libsl = LibSl::new();
        let ast = libsl.parse_file(
            diff_paths(path, env!("CARGO_MANIFEST_DIR")).unwrap().display().to_string(),
            &contents
        );

        assert_ron_snapshot!(match &ast {
            Ok(file) => Ok(file.with_libsl(&libsl)),
            Err(e) => Err(e.to_string()),
        });
    });
}
