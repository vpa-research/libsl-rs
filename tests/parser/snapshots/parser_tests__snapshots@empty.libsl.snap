---
source: tests/parser/main.rs
input_file: tests/parser/test-data/empty.libsl
---
Ok(File(
  loc: Span(
    start: 0,
    len: 0,
    file: "/home/slowlime/projects/github/vpa-research/libsl-rs/tests/parser/test-data/empty.libsl",
    line: Some(1),
    col: None,
  ),
  header: None,
  decls: [],
))
